tests/db_tests/test_entities.py

```diff
@@ -76,3 +76,49 @@ def test_project_finance_change_table(seeded_test_client_rollback):
     db.session.add(project_finance_change)
     with pytest.raises(IntegrityError):
         db.session.flush()
+
+
+def test_geospatial_dim_table(seeded_test_client_rollback):
+    """
+    Tests GeospatialDim table is correctly populated with reference data from
+    resources/geospatial_dim csv seed data, including checking the number
+    of rows in the table and that all itl1 region code and names are valid
+    pairs.
+
+    This test should fail if any new data is added to the csv or the
+    existing data is mutated. If the test fails it will need to be updated
+    and the database will need to be reseeded with the updated reference data.
+    """
+
+    geospatial_data = ents.GeospatialDim.query.all()
+
+    assert len(geospatial_data) == 124
+
+    assert geospatial_data[0].postcode_prefix == "AB"
+    assert geospatial_data[0].itl1_region_code == "TLM"
+    assert geospatial_data[0].data_blob == {"itl1_region_name": "Scotland"}
+
+    assert geospatial_data[-1].postcode_prefix == "ZE"
+    assert geospatial_data[-1].itl1_region_code == "TLM"
+    assert geospatial_data[-1].data_blob == {"itl1_region_name": "Scotland"}
+
+    itl1_region_pairs = {
+        "TLC": "North East",
+        "TLD": "North West",
+        "TLE": "Yorkshire",
+        "TLF": "East Midlands",
+        "TLG": "West Midlands",
+        "TLH": "East",
+        "TLI": "London",
+        "TLJ": "South East",
+        "TLK": "South West",
+        "TLL": "Wales",
+        "TLM": "Scotland",
+        "TLN": "Northern Ireland",
+        "TLZ": "Non-ITL1 Region",
+    }
+
+    for row in geospatial_data:
+        region_code = row.itl1_region_code
+        region_name = row.data_blob["itl1_region_name"]
+        assert (region_code, region_name) in itl1_region_pairs.items()
```

Comment:
this is fine and does exactly what you need, sorry to be nit-picky but would it be clearer (and slightly more performant) to do 
`assert itl_region_pairs[region_code] == region name`
or a similar assertion using .get method?

Reply:
or to deal with even smaller data structures, you could re-query the DB table, but selecting only region_code and region_name in the query, combined with a distinct, so you will iterate through less rows. Might not be worth the extra effort though :)